RockPaperScissors/RockPaperScissors/Models/GameManager.swift

```diff
@@ -7,46 +7,107 @@
+        var currentTurnOwner: PlayerType = winner
+        var isGameOn: Bool = true
+        
+        while isGameOn {
+            guard let gameOptionNumber = inputGameOptionNumber(gameType: .mukChiPa, currentTurnOwner) else {
+                currentTurnOwner = .computer
+                continue
+            }
+            
+            computer.makeRandomHandShape()
+            guard let computerHandShape = computer.getCurrentHandShape() else { continue }
+            
+            switch gameOptionNumber {
+            case endOption:
+                isGameOn = false
+                print("게임종료")
+            default:
+                let userHandShape = handShapes[gameOptionNumber - 1]
+                guard let winner = getGameResult(computerHandShape, userHandShape).winner else {
+                    isGameOn = false
+                    showMukChiPaGameWinner(with: currentTurnOwner)
+                    return
+                }
+                isGameOn = !getGameResult(computerHandShape, userHandShape).isGameOn
+                user.changeHandShape(to: userHandShape)
+                currentTurnOwner = winner
+                showCurrentPlayerTurn(with: currentTurnOwner)
+            }
+        }
+    }
+    
+    private func getGameResult(_ computerHandShape: HandShape, _ userHandShape: HandShape) -> (isGameOn: Bool, resultMessage: String, winner: PlayerType?) {
+        let gameResult: GameResult = compareHandShapeWith(computerHandShape, userHandShape)
+        
+        switch gameResult {
+        case .draw:
+            return (true, "비겼습니다.", nil)
+        case .win:
+            return (false, "이겼습니다!", .person)
+        case .lose:
+            return (false, "졌습니다!", .computer)
+        }
     }
     
-    private func getGameResult(_ computerHandShape: HandShape, _ userHandShape: HandShape) -> (isGameOn: Bool, resultMessage: String) {
+    private func compareHandShapeWith(_ computerHandShape: HandShape, _ userHandShape: HandShape) -> GameResult {
         switch (computerHandShape, userHandShape) {
         case let (computerHandShape, userHandShape) where computerHandShape == userHandShape:
-            return (true, "비겼습니다.")
+            return .draw
         case (.rock, .paper), (.paper, .scissors), (.scissors, .rock):
-            return (false, "이겼습니다!")
+            return .win
         default:
-            return (false, "졌습니다!")
+            return .lose
         }
     }
     
-    private func inputGameOptionNumber() -> Int? {
+    private func inputGameOptionNumber(gameType: GameType, _ currentTurnOwner: PlayerType? = nil) -> Int? {
         let validOptionRange: ClosedRange<Int> = 0...3
-        print("가위(1), 바위(2), 보(3)! <종료 : 0> :", terminator: " ")
+        showGameMessageWith(gameType, currentTurnOwner)
+        
         guard let gameOption = readLine(),
               let gameOptionNumber = Int(gameOption),
               validOptionRange.contains(gameOptionNumber) else {
```

Comment:
```suggestion
              0...3 ~= gameOptionNumber else {
```
이렇게도 사용할 수 있을 것 같아요👍🏼

Reply:
해리가 의견내주신 부분의 코드도 고민해보았습니다! 한 번 더 상기하게 되네요..! 감사합니다 😇